### PR TITLE
Notes for Apple M1 build from source

### PR DESCRIPTION
**Pull request by bitkarrot**
Installation notes for building from source for Apple M1 (ARM) 
@SomberNight

### CODE REVIEW

File: contrib/osx/notes_for_arm.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Notes on building Electrum on Apple M1 OSX (ARM)
```
Comment (SomberNight) [inline contrib/osx/notes_for_arm.md:1]:
```suggestion
# Notes on running Electrum from source on ARM-based Macs (Apple M1 OSX)
```

Looks like the title is not technically correct - the text is re how to run from source on apple m1.
Building suggests it is about creating binaries (such as the .dmg files we distribute).

File: contrib/osx/notes_for_arm.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+```
+python3 -m pip install --user -e .
+```
+
+2. pycryptodomex and cryptography and will need to be manually installed:
+
+```
+$ pip install pycryptodomex
+$ pip install cryptography
+```
```
Comment (SomberNight) [inline contrib/osx/notes_for_arm.md:23]: You don't need both, do you?
Either one should be enough; they are alternatives to each other.
Assuming that's so, we prefer `cryptography` (or at least let's be consistent with the top-level README.rst).

File: contrib/osx/notes_for_arm.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+3. Manually install libsecp256k1 from source (https://github.com/bitcoin-core/secp256k1#build-steps) but with the following modifications: 
+
+```
+$ ./autogen.sh
+$ ./configure --enable-module-recovery
+$ make
+$ make check  # run the test suite
+$ sudo make install  # optional
+```
```
Comment (SomberNight) [inline contrib/osx/notes_for_arm.md:33]: Does [`contrib/make_libsecp256k1.sh`](https://github.com/spesmilo/electrum/blob/cb39bbbd943e19b392302a1ac557c374fcd2d36b/contrib/make_libsecp256k1.sh) not work?

File: contrib/osx/notes_for_arm.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+$ echo 'export PATH="/opt/homebrew/opt/qt@5/bin:$PATH"' >> ~/.zshrc
+$ echo 'export PATH="/opt/homebrew/opt/pyqt@5/5.15.4_1/bin:$PATH"' >> ~/.zshrc
+$ source ~/.zshrc
+```
+
+Finally, try it in python to ensure it works: 
+
+```
+$ python3
+>> import PyQt5
```
Comment (SomberNight) [inline contrib/osx/notes_for_arm.md:48]:
```suggestion
>>> import PyQt5
```

File: contrib/osx/notes_for_arm.md (hunk 2 above)
Comment (bitkarrot) [reply contrib/osx/notes_for_arm.md:23]: Ok lets just make it `cryptography`

File: contrib/osx/notes_for_arm.md (hunk 3 above)
Comment (bitkarrot) [reply contrib/osx/notes_for_arm.md:33]: oops didn't see that. lets edit it to be explicit. Sure makes the next person's life a lot easier.

**Comment by bitkarrot** [comment]
let me know if anything else

Thanks @SomberNight

**Comment by SomberNight** [comment]
I think most of this is covered by the [README](https://github.com/spesmilo/electrum/blob/ad41f4aed01a8848ce9139f22f84a433c3b7c3cc/README.rst), except for the brew PyQt5 workaround, but for that I will merge it.
Thanks.

**Pull request merged**